core/src/main/java/io/questdb/cairo/RebuildIndex.java

```diff
@@ -0,0 +1,299 @@
+                                    tempStringSink,
+                                    partitionTimestamp,
+                                    partitionSize);
+                        }
+                    }
+                } else {
+                    long partitionSize = txReader.getTransientRowCount();
+                    rebuildIndex(rebuildColumnIndex, ff, indexer, metadata, partitionDirFormatMethod, tempStringSink, 0, partitionSize);
+                }
+            }
+        } finally {
+            metadata.close();
+            indexer.clear();
+            path.trimTo(rootLen);
+            lockName(path);
+            releaseLock(ff);
+        }
+    }
+
+    private void rebuildIndex(
+            int rebuildColumnIndex,
+            FilesFacade ff,
+            SymbolColumnIndexer indexer,
+            TableReaderMetadata metadata,
+            DateFormat partitionDirFormatMethod,
+            StringSink sink,
+            long partitionTimestamp,
+            long partitionSize
+    ) {
+        sink.clear();
+        partitionDirFormatMethod.format(partitionTimestamp, null, null, sink);
+
+        if (rebuildColumnIndex == ALL) {
+            for (int columnIndex = metadata.getColumnCount() - 1; columnIndex > -1; columnIndex--) {
+                if (metadata.isColumnIndexed(columnIndex)) {
+                    CharSequence columnName = metadata.getColumnName(columnIndex);
+                    int indexValueBlockCapacity = metadata.getIndexValueBlockCapacity(columnIndex);
+                    rebuildIndex(indexer, columnName, sink, indexValueBlockCapacity, partitionSize, ff);
+                }
+            }
+        } else {
+            if (metadata.isColumnIndexed(rebuildColumnIndex)) {
+                CharSequence columnName = metadata.getColumnName(rebuildColumnIndex);
+                int indexValueBlockCapacity = metadata.getIndexValueBlockCapacity(rebuildColumnIndex);
+                rebuildIndex(indexer, columnName, sink, indexValueBlockCapacity, partitionSize, ff);
+            } else {
+                throw CairoException.instance(0).put("Column is not indexed");
+            }
+        }
+    }
+
+    private void rebuildIndex(
+            SymbolColumnIndexer indexer,
+            CharSequence columnName,
+            CharSequence partitionName,
+            int indexValueBlockCapacity,
+            long partitionSize,
+            FilesFacade ff
+    ) {
+        path.trimTo(rootLen).concat(partitionName);
+        final int plen = path.length();
+
+        if (ff.exists(path.$())) {
+            try (final MemoryMR roMem = indexMem) {
+                removeIndexFiles(columnName, ff);
+                TableUtils.dFile(path.trimTo(plen), columnName);
+                LOG.info().$("indexing [path=").$(path).I$();
+
+                final long columnTop = TableUtils.readColumnTop(ff, path.trimTo(plen), columnName, plen, false);
+
+                if (partitionSize > columnTop) {
```

Comment:
when column is just added but has 0 rows this code will remove existing (empty) index and will not create any index files, not even initialized to 0